js/ui/gantt/ui.gantt.js

```diff
@@ -194,14 +197,108 @@ class Gantt extends Widget {
         this._$treeListWrapper.width(treeListWidth);
         this._$treeList.width(treeListWidth);
         this._$splitter.css("left", treeListWidth);
-        this._ganttView && this._ganttView._setWidth(this.$element().width() - treeListWidth - splitterBorderWidth);
+        this._ganttView && this._ganttView.setWidth(this.$element().width() - treeListWidth - splitterBorderWidth);
     }
 
     _setGanttViewOption(optionName, value) {
         this._ganttView && this._ganttView.option(optionName, value);
     }
+    _setTreeListOption(optionName, value) {
+        this._treeList && this._treeList.option(optionName, value);
+    }
+
+    _refreshDataSource(name) {
+        let dataOption = this[`_${name}Option`];
+        if(dataOption) {
+            dataOption._disposeDataSource();
+            delete this[`_${name}Option`];
+            delete this[`_${name}`];
+        }
+        if(this.option(`${name}.dataSource`)) {
+            dataOption = new DataOption(name, (name, data) => { this[`_${name}DataSourceChanged`](data); });
+            dataOption.option("dataSource", this.option(`${name}.dataSource`));
+            dataOption._refreshDataSource();
+            this[`_${name}Option`] = dataOption;
+        }
+    }
+    _tasksDataSourceChanged(data) {
+        const { keyExpr, parentIdExpr, startExpr, endExpr, progressExpr, titleExpr } = this.option("tasks"),
+            getId = dataCoreUtils.compileGetter(keyExpr),
```

Comment:
It would be nice to extract the common logic of generating getters and mapping *DataSourceChanged options